app/forms/backup_code_verification_form.rb

```diff
@@ -28,7 +28,7 @@ def backup_code_config
   def extra_analytics_attributes
     {
       multi_factor_auth_method: 'backup_code',
-      multi_factor_auth_method_created_at: backup_code_config&.created_at,
+      multi_factor_auth_method_created_at: backup_code_config&.created_at&.strftime('%s%L'),
```

Comment:
I would be in favor of changing this at the `Analytics` class level to deep iterate over hashes we send and convert all `Time` objects to this format, so that we don't have to remember at each callsite?

Reply:
Here's my version of a patch
```diff
diff --git a/app/services/analytics.rb b/app/services/analytics.rb
index bd7d1e245..1de876ed5 100644
--- a/app/services/analytics.rb
+++ b/app/services/analytics.rb
@@ -28,7 +28,7 @@ class Analytics
 
     analytics_hash.merge!(request_attributes) if request
 
-    ahoy.track(event, analytics_hash)
+    ahoy.track(event, deep_convert_timestamps(analytics_hash))
 
     # Tag NewRelic APM trace with a handful of useful metadata
     # https://www.rubydoc.info/github/newrelic/rpm/NewRelic/Agent#add_custom_attributes-instance_method
@@ -103,4 +103,15 @@ class Analytics
     return value unless value.is_a?(String)
     Time.zone.parse(value)
   end
+
+  def deep_convert_timestamps(payload)
+    payload.deep_transform_values do |value|
+      case value
+      when Time, DateTime
+        value.strftime('%s%L')
+      else
+        value
+      end
+    end
+  end
 end
```

 we could also do `.respond_to?(:strftime)` but otherwise same same

Reply:
makes sense ill update it

Reply:
that seems reasonable to me, I thought we'd confirmed that it works in the screenshots, so I was ready to commit

Reply:
> I think the primary issue for me is we're trading off a human-readable value for one that we can filter on, and if we don't plan to need filtering on it, we're losing the readability without much gain.

The query syntax would let you convert it back to human-readable if you wanted it, using [`fromMillis`](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/CWL_QuerySyntax-operations-functions.html#CWL_QuerySyntax-datetime).

At least my hunch is that for things other than the timestamp of the event itself, if we're logging a date, we're probably more interested in filtering on the value that we are to be looking at any specific individual value.